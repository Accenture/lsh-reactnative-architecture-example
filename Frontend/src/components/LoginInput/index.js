import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TextInput } from 'react-native';
import styles from './LoginInput.style';

const Input = ({
  placeholder,
  label,
  value,
  onChangeText,
  secureTextEntry,
}) => {
  const { inputStyle, labelStyle, containerStyle } = styles;
  return (
    <View style={containerStyle}>
      <Text style={labelStyle}>{label}</Text>
      <TextInput
        secureTextEntry={secureTextEntry}
        autoCapitalize="none"
        placeholder={placeholder}
        autoCorrect={false}
        value={value}
        onChange={onChangeText}
        style={inputStyle}
      />
    </View>
  );
};
Input.defaultProps = {
  placeholder: 'enter here',
  label: 'label',
  value: '',
  onChangeText: PropTypes.element.isRequired,
  secureTextEntry: false,
};

Input.propTypes = {
  placeholder: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.string,
  onChangeText: PropTypes.func,
  secureTextEntry: PropTypes.bool,
};
export default Input;
