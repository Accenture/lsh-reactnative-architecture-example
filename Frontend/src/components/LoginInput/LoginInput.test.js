import React from 'react';
import { shallow } from 'enzyme';
import LoginInput from './index';

describe('Tests login input component', () => {
  it('matches snapshot', () => {
    const comp = shallow(<LoginInput />);
    expect(comp).toMatchSnapshot();
  });
});
