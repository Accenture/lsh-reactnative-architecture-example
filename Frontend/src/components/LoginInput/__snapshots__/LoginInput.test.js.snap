// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests login input component matches snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
        "paddingLeft": 20,
      }
    }
  >
    label
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChange={[Function]}
    placeholder="enter here"
    secureTextEntry={false}
    style={
      Object {
        "color": "#000",
        "flex": 2,
        "fontSize": 18,
        "lineHeight": 23,
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
    value=""
  />
</Component>
`;
